### PR TITLE
Add setuptools command to build the man page

### PR DESCRIPTION
**Pull request by ana**
Add a new setuptools command to build the avocado manpage independently.

Reference: https://github.com/avocado-framework/avocado/issues/4372

### CODE REVIEW

**Comment by ana** [comment]
This change would require docutils  installed. Any suggestions about what's the best way to handle this?

**Comment by willianrampazzo** [comment]
> This change would require docutils installed. Any suggestions about what's the best way to handle this?

There is a `requirements-selftests.txt`, what if we create a new `requirements-docs.txt` and list everything we need for documentation build there?

**Comment by beraldoleal** [comment]
> > This change would require docutils installed. Any suggestions about what's the best way to handle this?
> 
> There is a `requirements-selftests.txt`, what if we create a new `requirements-docs.txt` and list everything we need for documentation build there?

I like the idea, my only suggestion would be to be "development" so:

  - requirements.txt : Everything that you need to run Avocado;
  - requirements-dev.txt: Everything that you need to HACK Avocado, including selftests and building docs

**Review by beraldoleal — COMMENTED** [review]
As mentioned by @willianrampazzo , we could track the requirements for hacking, somehow.

**Comment by willianrampazzo** [comment]
> > > This change would require docutils installed. Any suggestions about what's the best way to handle this?
> > 
> > 
> > There is a `requirements-selftests.txt`, what if we create a new `requirements-docs.txt` and list everything we need for documentation build there?
> 
> I like the idea, my only suggestion would be to be "development" so:
> 
> * requirements.txt : Everything that you need to run Avocado;
> * requirements-dev.txt: Everything that you need to HACK Avocado, including selftests and building docs

Oh, I thought about one specific requirement file for documentation thinking on those people that like to read a man page, for example, but will not hack Avocado. So, maybe move the docs requirements to the first file you suggested?

**Comment by beraldoleal** [comment]
> > > > This change would require docutils installed. Any suggestions about what's the best way to handle this?
> > > 
> > > 
> > > There is a `requirements-selftests.txt`, what if we create a new `requirements-docs.txt` and list everything we need for documentation build there?
> > 
> > 
> > I like the idea, my only suggestion would be to be "development" so:
> > 
> > * requirements.txt : Everything that you need to run Avocado;
> > * requirements-dev.txt: Everything that you need to HACK Avocado, including selftests and building docs
> 
> Oh, I thought about one specific requirement file for documentation thinking on those people that like to read a man page, for example, but will not hack Avocado. So, maybe move the docs requirements to the first file you suggested?

Do you think people that will need to read the man page, will clone/install requirements/build or just search it online/system?

IMO, for this case, we have only users and devs/contributors, so we would be fine with only two requirements.

**Comment by willianrampazzo** [comment]
> > > > > This change would require docutils installed. Any suggestions about what's the best way to handle this?
> > > > 
> > > > 
> > > > There is a `requirements-selftests.txt`, what if we create a new `requirements-docs.txt` and list everything we need for documentation build there?
> > > 
> > > 
> > > I like the idea, my only suggestion would be to be "development" so:
> > > 
> > > * requirements.txt : Everything that you need to run Avocado;
> > > * requirements-dev.txt: Everything that you need to HACK Avocado, including selftests and building docs
> > 
> > 
> > Oh, I thought about one specific requirement file for documentation thinking on those people that like to read a man page, for example, but will not hack Avocado. So, maybe move the docs requirements to the first file you suggested?
> 
> Do you think people that will need to read the man page, will clone/install requirements/build or just search it online/system?

I really don't know the answer to that. Personally speaking, I would go to the web documentation. But we have an example in the team :) one of the first things I saw @ana doing was updating the man documentation. TBH, I didn't even know Avocado had man pages rsrs.

Anyway, that was just a suggestion, we can check with others what they think. No strong opinion on my side.

> 
> IMO, for this case, we have only users and devs/contributors, so we would be fine with only two requirements.

**Comment by ana** [comment]
My goal with this PR was to migrate "make man" towards an action with setuptools (`python setup.py man`) and also a first step to install the manpage with `python setup.py install ` or `pip`.
Thanks for your comments @willianrampazzo , it made me to realize that installing the manpage with setuptools isn't worth it and it'll be probably a source of problems. People installing from source will be already reading the web documentation. So let's keep the current status where the manpage is only installed with the RPM packages.
 
I'm going to update the PR with only the first commit.

**Review by beraldoleal — APPROVED** [review]
LGTM, thank you.

**Review by willianrampazzo — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -167,7 +167,3 @@ propagate-version:
 	done
 
 .PHONY: source install clean check variables
-
-# implicit rule/recipe for man page creation
-%.1: %.rst
-	$(RST2MAN) $< $@
```
Comment (clebergnu) [inline Makefile:173]: Wouldn't it be better to move the `$(PYTHON) setup.py man` rule here and keep the prerequisite `man/avocado.1`? This way the man page would need to be regenerated if the respective `rst` is changed.

**Review by clebergnu — COMMENTED** [review]
Hi @ana , this looks good, but I have a question for you before my ack.  Thanks!

File: Makefile (hunk 1 above)
Comment (ana) [reply Makefile:173]: > Hi @ana , this looks good, but I have a question for you before my ack. Thanks!

Thanks @clebergnu, I made an update to the PR.

**Review by clebergnu — APPROVED** [review]
LGTM, thanks!

**Pull request merged**